.github/workflows/docker_build_main.yml

```diff
@@ -0,0 +1,46 @@
+name: Build Docker Images on Main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  build-and-push-images:
+    name: "Build and push Docker images with commit hash"
+    runs-on: warp-ubuntu-2204-x64-8x
+    permissions:
+      contents: read
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@5e57cd118135c172c3672efd75eb46360885c0ef # v3.6.0
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Install repro-env
+        run: |
+          wget 'https://github.com/kpcyrd/repro-env/releases/download/v0.4.3/repro-env'
+          echo '2a00b21ac5e990e0c6a0ccbf3b91e34a073660d1f4553b5f3cda2b09cc4d4d8a  repro-env' | sha256sum -c -
+          sudo install -m755 repro-env -t /usr/bin
+
+      - name: Install skopeo
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y skopeo
+
+      - name: Build and push both images
+        run: |
+          export LAUNCHER_IMAGE_NAME=mpc-launcher
+          ./deployment/build-images.sh --node --launcher --push
```

Comment:
nit: as we spoke, adding the `NODE_IMAGE_NAME` export

Reply:
Clean up tag from build script: https://github.com/near/mpc/pull/1460/commits/a480a89ed534fabe99078fee62b190062965f358

Reply:
And adding the node image name export: https://github.com/near/mpc/pull/1460/commits/28be106f18e85ae4a6cf7e616614d24c49d488e1